asset/src/count_by_field/interfaces.ts

```diff
@@ -6,6 +6,5 @@ export interface CountByField {
 
 export interface CountByFieldConfig extends OpConfig {
     field: string;
-    metric_api_name: string;
```

Comment:
Is it necessary to remove this?  Does the new API support this option?  If so, lets leave it in the asset.

Reply:
This field was used to initialize the `job_metric_api` which was registered on `context.apis.executionContext.job_metric_api`. It gave the ability for other APIs to be used instead. I could default to using the API at `context.apis.foundation.promMetrics`, but if `metric_api_name` is defined use that API instead.

Reply:
`private metrics!: JobMetricsAPI;`
This type would have prevented any other API from working if it had been passed as the `metric_api_name`. We could replace it with a type with an interface with abstract `inc` and `addMetric` methods that all the APIs must implement.

Reply:
I thought it just allowed changing the default name of the metrics created.  I am probably misremembering or misunderstanding something.